addons/mail/static/src/core/web/chatter.js

```diff
@@ -149,31 +149,25 @@ export class Chatter extends Component {
                     name: this.props.webRecord?.data?.display_name || undefined,
                 });
             }
-            await this.load(this.props.threadId, [
+            await this.load(this.props.threadModel, this.props.threadId, [
                 "followers",
                 "attachments",
                 "suggestedRecipients",
             ]);
+            this.changeThread(this.props.threadModel, this.props.threadId);
             this.scrollPosition.restore();
         });
         onPatched(this.scrollPosition.restore);
         onWillUpdateProps((nextProps) => {
-            this.load(nextProps.threadId, ["followers", "attachments", "suggestedRecipients"]);
-            if (nextProps.threadId === false) {
-                this.state.composerType = false;
-            }
-            this.attachmentUploader.thread = this.threadService.getThread(
-                nextProps.threadModel,
-                nextProps.threadId
-            );
-            if (this.onNextUpdate) {
-                if (!this.onNextUpdate(nextProps)) {
-                    this.onNextUpdate = null;
-                }
-            }
-            if (nextProps.threadId) {
-                this.closeSearch();
+            if (this.env.chatter?.fetchOnWillUpdateProps) {
+                this.env.chatter.fetchOnWillUpdateProps = false;
+                this.load(nextProps.threadModel, nextProps.threadId, [
+                    "followers",
+                    "attachments",
+                    "suggestedRecipients",
+                ]);
             }
+            this.changeThread(nextProps.threadModel, nextProps.threadId);
```

Comment:
`changeThread` should not be called when the thread doesn't change (same model and id as current thread)